example/heatEquation2D/src/heatEquation2D.cpp

```diff
@@ -0,0 +1,199 @@
+
+#ifdef PNGWRITER_ENABLED
+#    include "writeImage.hpp"
+#endif
+
+#include <alpaka/alpaka.hpp>
+#include <alpaka/example/ExecuteForEachAccTag.hpp>
+
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+#include <iostream>
+#include <type_traits>
+#include <utility>
+
+//! Each kernel computes the next step for one point.
+//! Therefore the number of threads should be equal to numNodesX.
+//! Every time step the kernel will be executed numNodesX-times
+//! After every step the curr-buffer will be set to the calculated values
+//! from the next-buffer.
+//!
+//! In standard projects, you typically do not execute the code with any available accelerator.
+//! Instead, a single accelerator is selected once from the active accelerators and the kernels are executed with the
+//! selected accelerator only. If you use the example as the starting point for your project, you can rename the
+//! example() function to main() and move the accelerator tag to the function body.
+template<typename TAccTag>
+auto example(TAccTag const&) -> int
+{
+    if constexpr(std::is_same_v<TAccTag, alpaka::TagCpuSerial>)
+    {
+        return EXIT_SUCCESS;
+    }
+    // Set Dim and Idx type
+    using Dim = alpaka::DimInt<2u>;
+    using Idx = uint32_t;
+
+    // Define the accelerator
+    using Acc = alpaka::TagToAcc<TAccTag, Dim, Idx>;
+    std::cout << "Using alpaka accelerator: " << alpaka::getAccName<Acc>() << std::endl;
+
+    // Select specific devices
+    auto const platformHost = alpaka::PlatformCpu{};
+    auto const devHost = alpaka::getDevByIdx(platformHost, 0);
+    auto const platformAcc = alpaka::Platform<Acc>{};
+    auto const devAcc = alpaka::getDevByIdx(platformAcc, 0);
+
+    // simulation defines
+    // Parameters
+    constexpr alpaka::Vec<Dim, Idx> numNodes{512, 1024}; // {Y, X}
+    constexpr alpaka::Vec<Dim, Idx> haloSize{2, 2};
+    constexpr alpaka::Vec<Dim, Idx> extent = numNodes + haloSize;
+
+    constexpr uint32_t numTimeSteps = 10000;
+    constexpr double tMax = 0.001;
+    // x, y in [0, 1], t in [0, tMax]
+    constexpr double dx = 1.0 / static_cast<double>(extent[1] - 1);
+    constexpr double dy = 1.0 / static_cast<double>(extent[0] - 1);
+    constexpr double dt = tMax / static_cast<double>(numTimeSteps);
+
+    // Check the stability condition
+    constexpr double r = dt / std ::min(dx * dx, dy * dy);
+    if constexpr(r > 0.5)
+    {
+        std::cerr << "Stability condition check failed: dt/min(dx^2,dy^2) = " << r
+                  << ", it is required to be <= 0.5\n";
+        return EXIT_FAILURE;
+    }
+
+    // Initialize host-buffer
+    // This buffer will hold the current values (used for the next step)
+    auto uBufHost = alpaka::allocBuf<double, Idx>(devHost, extent);
+
+    // Accelerator buffer
+    using BufAcc = alpaka::Buf<Acc, double, Dim, Idx>;
+    BufAcc uNextBufAcc{alpaka::allocBuf<double, Idx>(devAcc, extent)};
+    BufAcc uCurrBufAcc{alpaka::allocBuf<double, Idx>(devAcc, extent)};
```

Comment:
why `auto` on line 79 and an explicit type here ?
```suggestion
    auto uNextBufAcc = alpaka::allocBuf<double, Idx>(devAcc, extent);
    auto uCurrBufAcc = alpaka::allocBuf<double, Idx>(devAcc, extent);
```

Reply:
Changed to auto, it was a copy paste relic from the 1D heat equation example